### PR TITLE
Remove lucene support

### PR DESCRIPTION
**Pull request by MiguelAHM**
* Remove Lucene
* Change some tests to use Elastic Search
* Disable those tests that are failing (and are not easy to fix) once the test has been changed to ES
* Remember to remove dir.fulltext.index (used for Lucene) and elastic search.enabled (used for enabling Lucene if ES is disabled)

### CODE REVIEW

File: whois-api/src/test/java/net/ripe/db/whois/api/rdap/WhoisRdapServiceTestIntegration.java [hunk 1]
```diff
@@ -55,21 +55,33 @@
 import static org.junit.jupiter.api.Assertions.assertThrows;
 import static org.junit.jupiter.api.Assertions.fail;
 
-@Tag("IntegrationTest")
+@Tag("ElasticSearchTest")
 @DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_EACH_TEST_METHOD)
```
Comment (maggarwal13) [inline whois-api/src/test/java/net/ripe/db/whois/api/rdap/WhoisRdapServiceTestIntegration.java:59]: Why tag needs to be changed?

File: whois-api/src/test/java/net/ripe/db/whois/api/rdap/WhoisRdapServiceTestIntegration.java [hunk 2]
```diff
@@ -55,21 +55,33 @@
...
+    private static final String METADATA_INDEX = "metadata_rdap_service";
     @Autowired
     TestWhoisLog queryLog;
 
 
     @BeforeAll
     public static void beforeClass() {
+        System.setProperty("elastic.whois.index", WHOIS_INDEX);
+        System.setProperty("elastic.commit.index", METADATA_INDEX);
+        System.setProperty("fulltext.search.max.results", "10");
```
Comment (maggarwal13) [inline whois-api/src/test/java/net/ripe/db/whois/api/rdap/WhoisRdapServiceTestIntegration.java:73]: There is WhoisRdapElasticServiceTestIntegration .. Can you just remove the  ES tests from here?

File: whois-api/src/test/java/net/ripe/db/whois/api/rdap/WhoisRdapServiceAclTestIntegration.java [hunk 3]
```diff
@@ -13,17 +15,34 @@
 
 import static org.junit.jupiter.api.Assertions.fail;
 
-@Tag("IntegrationTest")
+@Tag("ElasticSearchTest")
 @DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_EACH_TEST_METHOD)
```
Comment (maggarwal13) [inline whois-api/src/test/java/net/ripe/db/whois/api/rdap/WhoisRdapServiceAclTestIntegration.java:19]: Why this is required?

File: whois-api/src/test/java/net/ripe/db/whois/api/rdap/WhoisRdapServiceAclTestIntegration.java [hunk 4]
```diff
@@ -13,17 +15,34 @@
...
 
     @Autowired
     private IpResourceConfiguration ipResourceConfiguration;
     @Autowired
     private TestPersonalObjectAccounting testPersonalObjectAccounting;
 
+    @BeforeAll
+    public static void beforeClass() {
+        System.setProperty("elastic.whois.index", WHOIS_INDEX);
+        System.setProperty("elastic.commit.index", METADATA_INDEX);
```
Comment (maggarwal13) [inline whois-api/src/test/java/net/ripe/db/whois/api/rdap/WhoisRdapServiceAclTestIntegration.java:35]: Can you revert these changes?

File: whois-api/src/test/java/net/ripe/db/whois/api/rdap/AbstractRdapIntegrationTest.java [hunk 5]
```diff
@@ -17,24 +17,20 @@
 import static org.hamcrest.Matchers.containsString;
 import static org.hamcrest.Matchers.is;
 
-public abstract class AbstractRdapIntegrationTest extends AbstractIntegrationTest {
+public abstract class AbstractRdapIntegrationTest extends AbstractElasticSearchIntegrationTest {
```
Comment (maggarwal13) [inline whois-api/src/test/java/net/ripe/db/whois/api/rdap/AbstractRdapIntegrationTest.java:21]: Please revert these changes, ES has a seperate profile and we need docker to run them

**Review by maggarwal13 — CHANGES_REQUESTED** [review]
Please use proper test profile for ES

File: whois-api/src/test/java/net/ripe/db/whois/api/rdap/AbstractRdapIntegrationTest.java (hunk 5 above)
Comment (MiguelAHM) [reply whois-api/src/test/java/net/ripe/db/whois/api/rdap/AbstractRdapIntegrationTest.java:21]: Rdap uses ES.

File: whois-api/src/test/java/net/ripe/db/whois/api/rdap/WhoisRdapServiceTestIntegration.java (hunk 1 above)
Comment (maggarwal13) [reply whois-api/src/test/java/net/ripe/db/whois/api/rdap/WhoisRdapServiceTestIntegration.java:59]: No, we use different tests for ES, please check implementations for abstract class : AbstractElasticSearchIntegrationTest

File: whois-api/src/test/java/net/ripe/db/whois/api/elasticsearch/WhoisRdapQueryLimitTestIntegration.java [hunk 6]
```diff
@@ -1,35 +1,51 @@
-package net.ripe.db.whois.api.rdap;
```
Comment (MiguelAHM) [inline whois-api/src/test/java/net/ripe/db/whois/api/elasticsearch/WhoisRdapQueryLimitTestIntegration.java:1]: All test cases uses ES and are not covered by another ES test class.
So I prefer to move it into elasticsearch package instead of moving all the tests to WhoisRdapElasticServiceTestIntegration

File: whois-api/src/test/java/net/ripe/db/whois/api/rdap/WhoisRdapServiceTestIntegration.java (hunk 1 above)
Comment (MiguelAHM) [reply whois-api/src/test/java/net/ripe/db/whois/api/rdap/WhoisRdapServiceTestIntegration.java:54]: As we talk via Mattermost I compared this class with WhoisRdapElasticServiceTestIntegration and removed all test cases that already exist in the other class. So we don't mix Elastic search tests and common Rdap test in same class.

File: whois-api/src/test/java/net/ripe/db/whois/api/rdap/WhoisRdapServiceTestIntegration.java (hunk 2 above)
Comment (MiguelAHM) [reply whois-api/src/test/java/net/ripe/db/whois/api/rdap/WhoisRdapServiceTestIntegration.java]: Done

File: whois-api/src/test/java/net/ripe/db/whois/api/rdap/WhoisRdapServiceAclTestIntegration.java (hunk 4 above)
Comment (MiguelAHM) [reply whois-api/src/test/java/net/ripe/db/whois/api/rdap/WhoisRdapServiceAclTestIntegration.java]: Done

File: whois-api/src/test/java/net/ripe/db/whois/api/rdap/AbstractRdapIntegrationTest.java (hunk 5 above)
Comment (MiguelAHM) [reply whois-api/src/test/java/net/ripe/db/whois/api/rdap/AbstractRdapIntegrationTest.java]: done

File: whois-api/src/main/java/net/ripe/db/whois/api/autocomplete/ElasticAutocompleteSearch.java [hunk 7]
```diff
@@ -32,9 +30,11 @@
 import java.util.stream.Collectors;
 
 @Component
-@Conditional(ElasticSearchCondition.class)
 public class ElasticAutocompleteSearch implements AutocompleteSearch {
 
+    public static final String OBJECT_TYPE_FIELD_NAME = "object-type";
+    public static final String PRIMARY_KEY_FIELD_NAME = "primary-key";
+    public static final String LOOKUP_KEY_FIELD_NAME = "lookup-key";
```
Comment (maggarwal13) [inline whois-api/src/main/java/net/ripe/db/whois/api/autocomplete/ElasticAutocompleteSearch.java:37]: These constant should be defined in ElasticIndexService may be? As we are also using them in ElasticIndexService and there is bean injection of ElasticIndexService here or create a constant  class for them?

File: whois-api/src/test/java/net/ripe/db/whois/api/fulltextsearch/FullTextSearchNoIndexTestIntegration.java [hunk 8]
```diff
@@ -15,11 +12,6 @@
 @Tag("IntegrationTest")
 public class FullTextSearchNoIndexTestIntegration extends AbstractIntegrationTest {
```
Comment (maggarwal13) [inline whois-api/src/test/java/net/ripe/db/whois/api/fulltextsearch/FullTextSearchNoIndexTestIntegration.java:13]: Can you shift these two tests in FulltextSearch integartaion class?

File: whois-api/src/main/java/net/ripe/db/whois/api/autocomplete/ElasticAutocompleteSearch.java (hunk 7 above)
Comment (MiguelAHM) [reply whois-api/src/main/java/net/ripe/db/whois/api/autocomplete/ElasticAutocompleteSearch.java:37]: I am ok moving those constant to ElasticIndexService, there are already more there

File: whois-api/src/test/java/net/ripe/db/whois/api/httpserver/RewriteEngineTestIntegration.java [hunk 9]
```diff
@@ -266,18 +250,6 @@ public void cors_preflight_request() {
         assertThat(response.getStatus(), is(HttpStatus.OK_200));
     }
 
-    @Test
-    public void fulltext_search() {
-        fullTextIndex.rebuild();
-
-        Response response = RestTest.target(getPort(), "fulltextsearch/select?facet=true&format=xml&hl=true&q=(test)&start=0&wt=json")
-                .request()
```
[the remark this answers is not in the document]
Comment (MiguelAHM) [reply whois-api/src/test/java/net/ripe/db/whois/api/httpserver/RewriteEngineTestIntegration.java:274]: I will move it to ElasticFullTextSearchTestIntegration

**Review by maggarwal13 — APPROVED** [review]

File: whois-api/src/test/java/net/ripe/db/whois/api/elasticsearch/ElasticAutocompleteServiceTestIntegration.java [hunk 10]
```diff
@@ -464,6 +477,15 @@ public void filter_comment_multiline_value() {
         assertThat(getValues(query("AA1", "organisation", "org-name"), "org-name"), contains("Any"));
     }
 
+    @Test
+    @Disabled
```
Comment (eshryane) [inline whois-api/src/test/java/net/ripe/db/whois/api/elasticsearch/ElasticAutocompleteServiceTestIntegration.java:481]: Add a `TODO` within the annotation explaining why it's disabled, e.g.
```
@Disabled("TODO: [MH] migrate from Lucene")
```

File: whois-api/src/test/java/net/ripe/db/whois/api/elasticsearch/WhoisRdapQueryLimitTestIntegration.java [hunk 11]
```diff
@@ -241,8 +257,9 @@ public void search_entity_organisation_by_handle_with_wildcard_prefix_middle_and
     }
 
     @Test
+    @Disabled
```
[the remark this answers is not in the document]
Comment (MiguelAHM) [reply whois-api/src/test/java/net/ripe/db/whois/api/elasticsearch/WhoisRdapQueryLimitTestIntegration.java]: @Disabled("TODO: [MH] migrate from Lucene. org-name search doesn't match 'test(2)' from test(2) organisation")

File: whois-api/src/test/java/net/ripe/db/whois/api/elasticsearch/ElasticAutocompleteServiceTestIntegration.java (hunk 10 above)
Comment (MiguelAHM) [reply whois-api/src/test/java/net/ripe/db/whois/api/elasticsearch/ElasticAutocompleteServiceTestIntegration.java]: @Disabled("TODO: [MH] migrated from Lucene, issue with ':' in ES")
